packages/profiling-node/src/utils.ts

```diff
@@ -1,19 +1,37 @@
-import * as os from 'node:os';
-import { env, versions } from 'node:process';
-import { isMainThread, threadId } from 'node:worker_threads';
-import type { Client, Context, Envelope, Event, StackFrame, StackParser } from '@sentry/types';
-
-import { GLOBAL_OBJ, forEachEnvelopeItem, logger } from '@sentry/utils';
-
+/* eslint-disable max-lines */
+import * as os from 'os';
+import type {
+  Client,
+  Context,
+  DebugImage,
+  DsnComponents,
+  Envelope,
+  Event,
+  EventEnvelopeHeaders,
+  Profile,
+  ProfileChunk,
+  ProfileChunkEnvelope,
+  SdkInfo,
+  StackFrame,
+  StackParser,
+  ThreadCpuProfile,
+} from '@sentry/types';
+import { GLOBAL_OBJ, createEnvelope, dsnToString, forEachEnvelopeItem, logger, uuid4 } from '@sentry/utils';
+
+import { env, versions } from 'process';
+import { isMainThread, threadId } from 'worker_threads';
+
+import type { ProfileChunkItem } from '@sentry/types/build/types/envelope';
+import type { ContinuousThreadCpuProfile } from '../../types/src/profiling';
 import { DEBUG_BUILD } from './debug-build';
-import type { Profile, RawThreadCpuProfile, ThreadCpuProfile } from './types';
-import type { DebugImage } from './types';
+import type { RawChunkCpuProfile, RawThreadCpuProfile } from './types';
 
 // We require the file because if we import it, it will be included in the bundle.
 // I guess tsc does not check file contents when it's imported.
 const THREAD_ID_STRING = String(threadId);
 const THREAD_NAME = isMainThread ? 'main' : 'worker';
 const FORMAT_VERSION = '1';
+const CONTINUOUS_FORMAT_VERSION = '2';
```

Comment:
This technically means we are pinned to a self-hosted version for this.

Once this is ready, we'll have to remember to come back to write a comment about what self-hosted version the format is compat with (and update the docs accordingly)

Let's please make sure the tracking issue has this noted down somewhere.

Reply:
oh yes. I added a todo item in https://github.com/getsentry/sentry-javascript/issues/12330 under the doc section!